### PR TITLE
msgpack: add module declaration

### PR DESCRIPTION
**Pull request by yohei1126**
- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:https://github.com/DefinitelyTyped/DefinitelyTyped/issues/14180

### CODE REVIEW

File: msgpack/index.d.ts [hunk 1]
```diff
@@ -81,3 +81,7 @@ interface MsgPackCallbackResult {
 }
 
 declare var msgpack: MsgPackStatic;
+
+declare module "msgpack" {
```
Comment (DanielRosenwasser) [inline msgpack/index.d.ts]: The appropriate fix is to replace this with

```ts
export = msgpack;
```

If `msgpack` is also available in as a global in the browser (it doesn't look like it is, but I'm not sure), then it'd be

```ts
export = msgpack;
export as namespace
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by bowdenk7** [comment]
@yohei1126 can you address the comments and refresh the PR?

**Comment by yohei1126** [comment]
fix it. please check.

**Comment by yohei1126** [comment]
sorry, it broke the build.

**Comment by zhengbli** [comment]
@yohei1126 any updates on this one?

**Comment by yohei1126** [comment]
@DanielRosenwasser  Do you know the solution for this build failure? 
https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/198251240

```
Error in msgpack
Command failed: node /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/tester/../../typescript-installs/2.0/node_modules/typescript/lib/tsc.js 
msgpack-tests.ts(3,14): error TS2304: Cannot find name 'msgpack'.
msgpack-tests.ts(4,1): error TS2304: Cannot find name 'msgpack'.
msgpack-tests.ts(6,20): error TS2304: Cannot find name 'msgpack'.
msgpack-tests.ts(7,1): error TS2304: Cannot find name 'msgpack'.
msgpack-tests.ts(15,41): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(16,40): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(16,69): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(18,45): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(18,74): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(19,1): error TS2304: Cannot find name 'msgpack'.
msgpack-tests.ts(24,41): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(25,40): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(25,71): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(27,44): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(27,75): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(28,1): error TS2304: Cannot find name 'msgpack'.
Error: There was a test failure.
    at Object.<anonymous> (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/tester/test-runner.ts:90:9)
    at next (native)
    at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/tester/test-runner.js:4:58)
    at process._tickCallback (internal/process/next_tick.js:103:7)
npm ERR! Test failed.  See above for more details.
```

**Comment by DanielRosenwasser** [comment]
Yes, `msgpack` is being used as a global.

So either you'll need to import `msgpack` from those files:

```ts
import msgpack = require("msgpack");
```

or if it truly can be used as a global (i.e. in the browser, you can just write `msgpack` and it'll work), then you'll need to add

```ts
export as namespace msgpack;
```

to the bottom of your file.

**Comment by yohei1126** [comment]
thanks! I will try it later.

**Comment by yohei1126** [comment]
there is still the same issue.

```
Testing msgpack
	Running: node /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/tester/../../typescript-installs/2.0/node_modules/typescript/lib/tsc.js 
	msgpack-tests.ts(15,41): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(16,40): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(16,69): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(18,45): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(18,74): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(24,41): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(25,40): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(25,71): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(27,44): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(27,75): error TS2304: Cannot find name 'MsgPackCallbackResult'.
	Running: node /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/bin/tester/../../typescript-installs/2.1/node_modules/typescript/lib/tsc.js 
	msgpack-tests.ts(15,41): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(16,40): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(16,69): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(18,45): error TS2304: Cannot find name 'MsgPackUploadOption'.
msgpack-tests.ts(18,74): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(24,41): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(25,40): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(25,71): error TS2304: Cannot find name 'MsgPackCallbackResult'.
msgpack-tests.ts(27,44): error TS2304: Cannot find name 'MsgPackDownloadOption'.
msgpack-tests.ts(27,75): error TS2304: Cannot find name 'MsgPackCallbackResult'.
```

**Comment by jramsay** [comment]
@DanielRosenwasser: the following works but I'm not sure it is the correct guidance.  I've seen this pattern used elsewhere - is the correct fix here to wrap all interfaces in index.d.ts with:

 declare namespace msgpack { ...}

and then at the bottom of the file:

declare var msgpack: msgpack.MsgPackStatic;
export = msgpack;
export as namespace msgpack;

Then in msgpack-tests.ts prepend namespace interfaces with msgpack.
ex:   before: (xhr: XMLHttpRequest, option: msgpack.MsgPackUploadOption) => { },

File: msgpack/index.d.ts [hunk 2]
```diff
@@ -81,3 +81,6 @@ interface MsgPackCallbackResult {
 }
 
 declare var msgpack: MsgPackStatic;
```
Comment (DanielRosenwasser) [inline msgpack/index.d.ts]: Correct thing to do here is wrap the entire thing in a namespace named `msgpack`, then keep this variable around with the following type:

```ts
declare var msgpack: msgpack.MsgPackStatic;
```

and then everything should work again.

**Comment by yohei1126** [comment]
@DanielRosenwasser All checks on CI  has passed. Could you please review the code again.

File: msgpack/index.d.ts (hunk 2 above)
Comment (ogis-onishi) [inline msgpack/index.d.ts]: I fixed it. Please check.

**Comment by sandersn** [comment]
I believe @DanielRosenwasser is attending a conference right now and will be able to review the new changes in a few days.

**Comment by andy-ms** [comment]
We've recently updated the repository structure.
All packages are now in a `types` subdirectory.
Please rebase this pull request onto the latest `master` branch.

**Comment by yohei1126** [comment]
@andy-ms  Thanks. I did it.

**Comment by ogis-onishi** [comment]
@DanielRosenwasser Could you check it today if possible?

**Comment by yohei1126** [comment]
@DanielRosenwasser Could you check it today if possible?

**Comment by sandersn** [comment]
Looks good to me.

**Pull request merged**